### DISCUSSION TITLE
Is it possible to have a textarea with autogrow but only up to a specific number of lines (or pixels)?

### DISCUSSION THREAD

**Discussion by thikade** [zauberzeug/nicegui#3283] [Q&A] [answered]
### Question

Hi,
```py
ui.textarea().props("autogrow")
```
works nicely, but will grow infinitely. I would like to create a textarea that grows up to a set number of lines (or maxHeight pixels), and then stops growing, displaying a scrollbar instead. Just as it does when you set the number of rows to a fixed number:
```py
ui.textarea().props("rows=3") 
```
Is this possible?

**Comment by falkoschindler** [comment, marked as answer]
Hi @thikade,

We can combine the "autogrow" prop with a maximum height:

```py
ui.textarea().props('autogrow input-class=max-h-14')
```

In its current version "min-h" or "max-h" doesn't support pixels (#3176), but we can use CSS instead:
```py
ui.textarea().props('autogrow input-style="max-height: 60px"')
```

**Reply by thikade** [reply]
Awesome, thanks so much for this awesome library - and replying this quickly!
